src/Crate/PDO/Http/ServerPool.php

```diff
@@ -0,0 +1,227 @@
+                continue;
+
+            } catch (BadResponseException $exception) {
+
+                $body = (string)$exception->getResponse()->getBody();
+                $json = json_decode($body, true);
+
+                if ($json === null && json_last_error() !== JSON_ERROR_NONE) {
+                    throw new RuntimeException(sprintf('Server returned non-JSON response: %s', $body), 0, $exception);
+                }
+
+                $errorCode    = $json['error']['code'];
+                $errorMessage = $json['error']['message'];
+
+                throw new RuntimeException($errorMessage, $errorCode, $exception);
+            }
+        }
+
+        throw new ConnectException(
+            sprintf('No more servers available, exception from last server: %s', $exception->getMessage()),
+            $exception->getRequest(),
+            $exception
+        );
+    }
+
+    /**
+     * {@Inheritdoc}
+     */
+    public function getServerInfo(): array
+    {
+        return [
+            'serverVersion' => $this->getServerVersion(),
+        ];
+    }
+
+    /**
+     * {@Inheritdoc}
+     */
+    public function getServerVersion(): string
+    {
+        $result = $this->execute("select version['number'] from sys.nodes limit 1", []);
+
+        if (count($result->getRows()) !== 1) {
+            throw new RuntimeException('Failed to determine server version');
+        }
+
+        return $result->getRows()[0][0];
+    }
+
+    /**
+     * Determines the options to pass to guzzle for each request
+     *
+     * @param string $server
+     * @param string $query
+     * @param array  $parameters
+     *
+     * @return array
+     */
+    private function getHttpOptions(string $server, string $query, array $parameters): array
+    {
+        $sslMode = $this->pdo->getAttribute(PDO::CRATE_ATTR_SSL_MODE);
+
+        $protocol = $sslMode === PDO::CRATE_ATTR_SSL_MODE_DISABLED ? 'http' : 'https';
+
+        $options = [
+            'base_uri'                      => sprintf('%s://%s', $protocol, $server),
+            RequestOptions::TIMEOUT         => $this->pdo->getAttribute(PDO::ATTR_TIMEOUT),
+            RequestOptions::CONNECT_TIMEOUT => $this->pdo->getAttribute(PDO::ATTR_TIMEOUT),
+            RequestOptions::AUTH            => $this->pdo->getAttribute(PDO::CRATE_ATTR_HTTP_BASIC_AUTH) ?: null,
+            RequestOptions::JSON            => [
+                'stmt' => $query,
+                'args' => $parameters,
+            ],
+
+            RequestOptions::HEADERS => [
+                'Default-Schema' => $this->pdo->getAttribute(PDO::CRATE_ATTR_DEFAULT_SCHEMA),
+            ],
+        ];
+
+        if ($sslMode === PDO::CRATE_ATTR_SSL_MODE_ENABLED) {
+            $options['verify'] = false;
+        }
+
+        $ca         = $this->pdo->getAttribute(PDO::CRATE_ATTR_SSL_CA);
+        $caPassword = $this->pdo->getAttribute(PDO::CRATE_ATTR_SSL_CA_PASSWORD);
```

Comment:
`$caPassword` is only needed `if ($ca) {`. I suggest making these 3 blocks 3 `private` methods:

```php
return array_merge(
    $options,
    $this->certificateAuthorityOptions(),
    $this->certificateOptions(),
    $this->sslKeyOptions
);
```

Also: please don't abbreviate